### PR TITLE
BACK-1733 - API Tests tooling : detect end of sync thanks to published message

### PR DESCRIPTION
**Pull request by mbasire-ledger**
* The publisher is stubbed. No messages sent to Rabbitmq.
  * ETH tests moved to a specific test suite
  * assertCreateAccount waits for end of sync when expected status is 200
  * no need to force sync in tests anymore

  * ( docker-compose-deps.yml starts dependencies of the WD. )
  * ( Fix some lifecycle message sent from AccountOperationsPublisher )

### CODE REVIEW

File: src/it/scala/co/ledger/wallet/daemon/api/AccountsApi_BTC_Test.scala [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
 import com.fasterxml.jackson.databind.JsonNode
 import com.twitter.finagle.http.{Response, Status}
+import org.scalatest.concurrent.PatienceConfiguration.Timeout
+import org.scalatest.concurrent.ScalaFutures
 
+import scala.concurrent.duration._
 import java.util.UUID
 
-class AccountsApiTest extends APIFeatureTest {
+class AccountsApi_BTC_Test extends APIFeatureTest with ScalaFutures {
```
Comment (RomanWlm) [inline src/it/scala/co/ledger/wallet/daemon/api/AccountsApi_BTC_Test.scala:16]: 🐍

Comment (RomanWlm) [reply src/it/scala/co/ledger/wallet/daemon/api/AccountsApi_BTC_Test.scala:16]: Could we keep scala naming convention and use Camel casing ?

File: src/it/scala/co/ledger/wallet/daemon/api/AccountsApi_BTC_Test.scala [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
   val poolName = "account_api_test_pool"
 
   override def beforeAll(): Unit = {
     super.beforeAll()
     createPool(poolName)
     assertWalletCreation(poolName, poolName, "bitcoin", Status.Ok)
-    assertCreateAccount(CORRECT_BODY_BITCOIN, poolName, poolName, Status.Ok)
-    assertSyncAccount(poolName, poolName, 0)
+
+    assertCreateAccount(CORRECT_BODY_BITCOIN, poolName, poolName, Status.Ok, 0).futureValue(Timeout(30.seconds))
```
Comment (RomanWlm) [inline src/it/scala/co/ledger/wallet/daemon/api/AccountsApi_BTC_Test.scala:24]: What about include timeout into assertion method ?

**Review by RomanWlm — CHANGES_REQUESTED** [review]

**Comment by RomanWlm** [comment]
Seems good, can you link the tickets related to this please ?

File: src/it/scala/co/ledger/wallet/daemon/api/AccountsApi_BTC_Test.scala (hunk 1 above)
Comment (mbasire-ledger) [reply src/it/scala/co/ledger/wallet/daemon/api/AccountsApi_BTC_Test.scala:16]: Yes, i can user Camel case. But it is far less readable for tests reports.

File: src/it/scala/co/ledger/wallet/daemon/api/AccountsApi_BTC_Test.scala (hunk 2 above)
Comment (mbasire-ledger) [reply src/it/scala/co/ledger/wallet/daemon/api/AccountsApi_BTC_Test.scala:24]: > Seems good, can you link the tickets related to this please ?

BACK-1733

**Comment by mbasire-ledger** [comment]
> Seems good, can you link the tickets related to this please ?

BACK-1733

File: src/it/scala/co/ledger/wallet/daemon/api/AccountsApi_BTC_Test.scala (hunk 2 above)
Comment (mbasire-ledger) [reply src/it/scala/co/ledger/wallet/daemon/api/AccountsApi_BTC_Test.scala]: Ok, i will make assertCreateAccount a blocking call with a 10 seconds timeout.

**Review by twilgenbus-ledger — APPROVED** [review]
LGTM

**Comment by twilgenbus-ledger** [comment]
Remarks taken into account 👍  Ready to merge for me

**Pull request merged**